### PR TITLE
changing generate_api_warszawa_url and some paths

### PR DESCRIPTION
**Pull request by Maryl12342**
Hello @mgierm , @klepper24, please review my changes.

### CODE REVIEW

**Review by klepper24 — APPROVED** [review]
looks fine for me

File: orchestrator/dags/utils.py [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+    def _get(self, url, params=None):
+        if params is None:
+            params = {}
+        default_params = {"apikey": self._apikey}
+        default_params.update(params)
+        response = self._client.get(
+            f"{ApiWarsawUrl.BASE_URL}{url}", params=default_params
+        )
+        response.raise_for_status()
+        return response.status_code, response.json()
```
Comment (mgierm) [inline orchestrator/dags/utils.py]:
```suggestion
        response.raise_for_status()
        return response.json()
```
If there is a call for `raise_for_status`, there is rather no need to return the status code (uneless we really need to handle different 2xx statuses in the executed code). Take a look at the [source of `raise_for_status`](https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.raise_for_status)

File: orchestrator/dags/utils.py [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+
+    def get_dbstore(self, id=None):
+        params = {}
+        if id is not None:
+            params["id"] = id
+        self._dbstore = self._get(ApiWarsawUrl.ENDPOINT_DBSTORE, params)
+        return self._dbstore
+
+    def get_busestrams(self, resource_id=None, type=None):
```
Comment (mgierm) [inline orchestrator/dags/utils.py]: I suggest to rename parameter `type` to something different, like `resource_type` - `type` is a [built-in function](https://docs.python.org/3/library/functions.html#type), so better to not overwrite it.

File: orchestrator/dags/utils.py [hunk 3]
```diff
@@ -1,12 +1,38 @@
...
+        params = {}
+        if resource_id is not None and type is not None:
+            params["resource_id"] = resource_id
+            params["type"] = type
+        self._busestrams = self._get(ApiWarsawUrl.ENDPOINT_BUSESTRAMS, params)
+        return self._busestrams
```
Comment (mgierm) [inline orchestrator/dags/utils.py]:
```suggestion
        return self._get(ApiWarsawUrl.ENDPOINT_BUSESTRAMS, params)
        
```
The `_busestrams` is not used across the class, so we don't need to define it.

File: orchestrator/dags/utils.py [hunk 4]
```diff
@@ -1,12 +1,38 @@
...
-def generate_api_warszawa_url(api_key: str, resource_id: str, **kwargs):
-    """
-    Returns url to Warsaw Public Transport API with given parameters as query strings
-    """
-    url = settings.API_WARSZAWA_URL
-    url += f'?apikey={api_key}&resource_id={resource_id}'
-    for k, v in kwargs.items():
-        url += f'&{k}={v}'
-    return url
+class ApiWarsawUrl:
```
Comment (mgierm) [inline orchestrator/dags/utils.py]: Maybe just `ApiWarsaw`, or `WarsawApi`? :wink:

File: orchestrator/dags/delays_processing.py [hunk 5]
```diff
@@ -22,14 +22,11 @@
 
 
 def save_tram_gps(ti) -> None:
-    url = generate_api_warszawa_url(API_KEY, RESOURCE_ID, type=2)
-    r = requests.get(url)
-    status = r.status_code
+    api = ApiWarsawUrl(apikey=API_KEY)
+    status, json_response = api.get_busestrams(resource_id=RESOURCE_ID, type=2)
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py]: If you apply my suggestion to not return status code, then you could remove the following `if` statement.

File: orchestrator/dags/delays_processing.py [hunk 6]
```diff
@@ -73,21 +70,19 @@ def save_tram_gps(ti) -> None:
 
     spark_job = SparkSubmitOperator(
         task_id="spark_job",
-        application="/opt/spark/execution_scripts/delays_proc.py",  # Spark app path created in airflow & spark cluster
+        application="/opt/spark/execution_scripts/delays_proc.py", # Spark application path created in airflow and spark cluster
         application_args=[settings.TRAM_FOLDER],
         name="delays processing",
         conn_id="spark_default",
-        conf={"spark.master": settings.SPARK_MASTER},
+        conf={"spark.master":settings.SPARK_MASTER},
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py:77]:
```suggestion
        conf={"spark.master": settings.SPARK_MASTER},
```

File: orchestrator/dags/general_data_import.py [hunk 7]
```diff
@@ -181,8 +180,8 @@ def get_json_from_api(link: str) -> json:
         
 def create_stops_list() -> List[Stop]:
     stops = []
-    url = generate_api_warszawa_url(API_KEY, RESOURCE_ID)
-    json_string = get_json_from_api(url)
+    api = ApiWarsawUrl(apikey=API_KEY)
+    _, json_string = api.get_dbstore(id=RESOURCE_ID)
```
Comment (mgierm) [inline orchestrator/dags/general_data_import.py]: If you apply my suggestion to not return status code, then you could remove the `_` variable.

**Comment by Maryl12342** [comment]
@mgierm, @meg25ellele, @klepper24, you can take a look on my changes: 
4c23be333b59d512f0e36a894dcb7bc0192af595

File: orchestrator/dags/delays_processing.py [hunk 8]
```diff
@@ -73,21 +70,19 @@ def save_tram_gps(ti) -> None:
...
         packages="org.mongodb.spark:mongo-spark-connector_2.12:3.0.1,com.microsoft.sqlserver:mssql-jdbc:11.2.0.jre8",
         dag=dag)
         
     task_move_to_archive = BashOperator(
         task_id="move_to_archive",
-        bash_command=f'mkdir -p {settings.SPARK_OUT_DIR}/ARCHIVE/'
-                     f' & mv {settings.TRAM_FOLDER}/* {settings.SPARK_OUT_DIR}/ARCHIVE/'
-                     f' & rm -R {settings.TRAM_FOLDER}'
+        bash_command=f'mkdir -p {settings.SPARK_OUT_DIR}ARCHIVE/ & mv {settings.TRAM_FOLDER}* {settings.SPARK_OUT_DIR}ARCHIVE/ & rm -R {settings.TRAM_FOLDER}'
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py:83]: Reverted change, accidently I guess.

File: orchestrator/dags/delays_processing.py [hunk 9]
```diff
@@ -73,21 +70,19 @@ def save_tram_gps(ti) -> None:
...
     )        
     
     end = EmptyOperator(task_id="end", dag=dag)
 
-    start >> task_save_tram_gps >> spark_job >> task_move_to_archive >> end
+    start >> task_save_tram_gps >> spark_job >> task_move_to_archive >> end
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py:88]: Reverted change (no newline an the end of file), accidently I guess.

File: orchestrator/dags/delays_processing.py [hunk 10]
```diff
@@ -22,14 +22,11 @@
...
-    r = requests.get(url)
-    status = r.status_code
+    api = ApiWarsawUrl(apikey=API_KEY)
+    status, json_response = api.get_busestrams(resource_id=RESOURCE_ID, type=2)
     if status == HTTPStatus.OK:
-        json_response = r.json()
-
         # saving json to file
-        filename = f'{settings.TRAM_FOLDER}/tram{settings.TODAY_TIME}.json'
+        filename = f'{settings.TRAM_FOLDER}tram{settings.TODAY_TIME}.json'
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py]: Reverted change, accidently I guess.

File: orchestrator/dags/delays_processing.py [hunk 11]
```diff
@@ -73,21 +70,19 @@ def save_tram_gps(ti) -> None:
 
     spark_job = SparkSubmitOperator(
         task_id="spark_job",
-        application="/opt/spark/execution_scripts/delays_proc.py",  # Spark app path created in airflow & spark cluster
+        application="/opt/spark/execution_scripts/delays_proc.py", # Spark application path created in airflow and spark cluster
```
Comment (mgierm) [inline orchestrator/dags/delays_processing.py:73]: Reverted change (too many chars in a line), accidently I guess.

File: orchestrator/dags/general_data_import.py [hunk 12]
```diff
@@ -394,3 +394,5 @@ def load_routes_to_MongoDB(ti) -> None:
     start >> task_download_general_ztm_data >> task_extract_lines >> task_load_timetable_to_MongoDB \
         >> task_remove_files >> task_load_calendar_to_MongoDB >> task_load_stops_to_MongoDB \
         >> task_extract_routes_lines >> task_load_routes_to_MongoDB >> end
+    
+
```
Comment (mgierm) [inline orchestrator/dags/general_data_import.py]:
```suggestion

```

File: orchestrator/dags/settings.py [hunk 13]
```diff
@@ -5,17 +5,17 @@
 
 
 SPARK_MASTER = "spark://spark:7077"
-SPARK_OUT_DIR = '/opt/spark/execution_scripts'
+SPARK_OUT_DIR = '/opt/spark/execution_scripts/'
```
Comment (mgierm) [inline orchestrator/dags/settings.py:8]: Reverted change, accidently I guess.

File: orchestrator/dags/settings.py [hunk 14]
```diff
@@ -5,17 +5,17 @@
...
+
 
 NOW = datetime.now()
-TODAY_DATETIME = datetime.now().astimezone(tz.gettz('Europe/Warsaw'))
+TO_ZONE = tz.gettz('Europe/Warsaw')
+TODAY_DATETIME = datetime.now().astimezone(TO_ZONE)
```
Comment (mgierm) [inline orchestrator/dags/settings.py:13]: Reverted change, accidently I guess.

File: orchestrator/dags/settings.py [hunk 15]
```diff
@@ -5,17 +5,17 @@
...
 TODAY_DATE = TODAY_DATETIME.strftime("%Y%m%d")
 TODAY_TIME = TODAY_DATETIME.strftime("%H%M%S")
 
-TRAM_FOLDER = os.path.join(SPARK_OUT_DIR, TODAY_DATE)
-
-API_WARSZAWA_URL = 'https://api.um.warszawa.pl/api/action/busestrams_get/'
+TRAM_FOLDER = f'{SPARK_OUT_DIR}{TODAY_DATE}/'
```
Comment (mgierm) [inline orchestrator/dags/settings.py:17]: Reverted change, accidently I guess.

File: orchestrator/dags/utils.py [hunk 16]
```diff
@@ -1,12 +1,38 @@
...
+            params = {}
+        default_params = {"apikey": self._apikey}
+        default_params.update(params)
+        response = self._client.get(
+            f"{ApiWarsawUrl.BASE_URL}{url}", params=default_params
+        )
+        response.raise_for_status()
+        return response.status_code, response.json()
+
+    def get_dbstore(self, id=None):
```
Comment (mgierm) [inline orchestrator/dags/utils.py]: I suggest to rename parameter `id` to something different, like `resource_id` - `id` is a [built-in function](https://docs.python.org/3/library/functions.html#id), so better to not overwrite it.

File: orchestrator/dags/settings.py [hunk 17]
```diff
@@ -5,17 +5,17 @@
...
 
 ZTM_GENERAL_LINK = 'ftp://rozklady.ztm.waw.pl'
 
-MONGO_HOST = 'git_mongo-python_1'
+MONGO_HOST = 'git_mongo-python_1'
```
Comment (mgierm) [inline orchestrator/dags/settings.py:21]:
```suggestion
MONGO_HOST = 'git_mongo-python_1'

```
Reverted change, accidently I guess.

**Review by mgierm — CHANGES_REQUESTED** [review]
Good job generally :clap: 
I added few suggestions mostly about naming objects and some improvements.
I also noticed that few changes where reverted (pep8 fixes), so maybe you just forgot to pull last changes from `main` branch.

**Pull request merged**